balances-tx-payment-extension/src/balance_conversion.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2019 Alain Brenzikofer
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::{AssetBalanceOf, AssetIdOf, BalanceOf};
+use core::marker::PhantomData;
+use encointer_primitives::{balances::BalanceType, communities::CommunityIdentifier};
+use frame_support::traits::{fungibles, tokens::BalanceConversion};
+use pallet_encointer_balances::Pallet as BalancesPallet;
+use pallet_encointer_communities::{Config as CommunitiesConfig, Pallet as CommunitiesPallet};
+
+pub fn balance_to_community_balance<T: CommunitiesConfig>(
+	balance: u128,
+	cid: CommunityIdentifier,
+	reward: u128,
+	fee_conversion_factor: u32,
+	asset_balance_decimals: u8,
+) -> u128 {
+	// 5.233 micro ksm correspond to 0.01 units of the community currency assuming a feeConversionFactor of 10_000
+	// the KSM balance parameter comes with 12 decimals
+	// 5.233 * 10^6 pKSM = 0.01 * 10^decimals LEU
+	// 5.233 * 10^6 pKSM = 0.01 * 10^(decimals - 4) * feeConversionFactor LEU
+	// 1 pKSM = (0.01 * 10^(decimals - 4) * feeConversionFactor) / (5.233 * 10^6) LEU
+	// 1 pKSM = (0.01 * 10^(decimals - 10) * feeConversionFactor) / 5.233 LEU
+	let conversion_factor = ((0.01f64 / 5.233f64) *
+		10i128.pow((asset_balance_decimals - 10) as u32) as f64 *
+		fee_conversion_factor as f64) as u128;
+
+	// assuming a nominal income of 20
+	return (balance * conversion_factor * reward) /
+		BalancesPallet::<T>::balance_type_to_fungible_balance(
+			cid.into(),
+			BalanceType::from_num(20i32),
+		)
```

Comment:
I think this calculation is wrong, and should not incorporate any hardcoded numbers. Just do it as @brenzi said it should be:

```
CCfee = KSMfee * factor * reward(cid)
```

Reply:
I updated the meaning of the `FeeConversionFactor`

Reply:
One consideration here: whatever the formula, we should ensure that we have enough slack in both directions to adjust fees. I.e. if we define the factor as an u64 and initialize it to a value of 5 that would be bad because we can only lower fees by a factor of 5. Rather add a few zeros in that case

Reply:
i set the factor to 10000 in the genesis config and calculated the formula based on this factor and the constraints discussed with @brenzi . i think 10000 allows for quite some slack